### PR TITLE
Issue: Class AuditEntry Not Found

### PR DESCRIPTION
**Pull request by aydreeihn**
This commit fixes an issue where we were trying to use a Plugin function that some helpdesks may not include. Rather than calling the class from the Plugin, the Plugin needed to have a new Signal to handle auditing certain deletions.

### CODE REVIEW

**Comment by aydreeihn** [comment]
**Note:** We will need to merge the PR for the plugin as well with this one.

File: scp/categories.php [hunk 1]
```diff
@@ -124,15 +124,8 @@
...
-                                        && $data = AuditEntry::getDataById($id, 'C'))
-                                    $name = json_decode($data[2], true);
-                                else {
-                                    $name = __('NA');
-                                    $data = array('C', $id);
-                                }
-                                $type = array('type' => 'deleted');
-                                Signal::send('object.deleted', $data, $type);
+                                $data = array('C', true);
+                                Signal::send('object.audit', $id, $data);
```
Comment (protich) [inline scp/categories.php]: Why not send the signal `object.deleted`  plus the object in Category::delete() routine called in line 112 above?

Comment (protich) [reply scp/categories.php]: The advantage here is anyone (plugins) can catch such signal and do whatever with it. `object.audit` is just too specific to audit plugin.

File: scp/helptopics.php [hunk 2]
```diff
@@ -171,15 +171,8 @@
...
-                                        && $data = AuditEntry::getDataById($id, 'H'))
-                                    $name = json_decode($data[2], true);
-                                else {
-                                    $name = __('NA');
-                                    $data = array('H', $id);
-                                }
-                                $type = array('type' => 'deleted');
-                                Signal::send('object.deleted', $data, $type);
+                                $data = array('H', true);
+                                Signal::send('object.audit', $id, $data);
```
Comment (protich) [inline scp/helptopics.php]: Same as above

**Review by protich — COMMENTED** [review]
Let's talk about this PR when you get a moment.

File: include/class.page.php [hunk 3]
```diff
@@ -153,10 +153,11 @@ function delete() {
 
         if (!parent::delete())
             return false;
-
-        return Topic::objects()
-            ->filter(array('page_id'=>$this->getId()))
-            ->update(array('page_id'=>0));
```
Comment (aydreeihn) [inline include/class.page.php:159]: This didn't appear to be relevant because we do not allow the Agent to delete a Page if a Topic is using it.

Comment (protich) [reply include/class.page.php:159]: Why the need for else block below?

Comment (aydreeihn) [reply include/class.page.php:159]: I went ahead and took the else out.

**Pull request merged**